### PR TITLE
[luci-interpreter] Refactor kernel builder for configurable build

### PR DESCRIPTION
**Pull request by SlavikMIPT**
This commit refactors kernel builder for configurable build.

ONE-DCO-1.0-Signed-off-by: Vyacheslav Bazhenov <slavikmipt@gmail.com>

### CODE REVIEW

**Comment by SlavikMIPT** [comment]
@nnfw-bot test nncc-debug

**Comment by SlavikMIPT** [comment]
@nnfw-bot test nncc-release

File: compiler/luci-interpreter/src/loader/KernelBuilder.h [hunk 1]
```diff
@@ -30,17 +30,29 @@
 namespace luci_interpreter
 {
 
+#define REGISTER_KERNEL(name)                                                            \
```
Comment (binarman) [inline compiler/luci-interpreter/src/loader/KernelBuilder.h:33]: Do we need to expose these functions in header?

**Review by binarman — APPROVED** [review]
LGTM

**Review by BalyshevArtem — APPROVED** [review]
LGTM

**Pull request merged**